mlrun/projects/pipelines.py

```diff
@@ -582,6 +586,19 @@ def run(
             project.set_source(source=source)
 
         namespace = namespace or config.namespace
+
+        # fallback to old notification behavior
+        if notifications:
+            logger.warning(
+                "Setting notifications on kfp pipeline runner uses old notification behavior. "
+                "Notifications will only be sent if you wait for pipeline completion. "
+                "To use the new notification behavior, use the remote pipeline runner."
+            )
```

Comment:
I'm unsure how we handle deprecation warnings, but maybe this message should come with some timeline for when we expect to stop supporting the old strategy.

Reply:
not that I'm aware 😅 

Maybe I'm confused by what is the "old notification behavior". Is this something that'll stop working eventually?

Reply:
Currently no plan to stop this behaviour. We might in the future but as there's no current deprecation plan, It's not the time to mention it I think.

Reply:
Sounds good. Thanks, man!